File: hector_quadrotor/hector_quadrotor_navigation/src/quadrotor_navigation.cpp
#include <ros/ros.h>
#include <geometry_msgs/Pose.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <hector_uav_msgs/PoseAction.h>
#include <hector_uav_msgs/EnableMotors.h>

int main (int argc, char **argv)
{
  ros::init(argc, argv, "quadrotor_navigation");

  ros::NodeHandle n;

  ROS_INFO("Enabling motors for hector_quadrotor");
  ros::ServiceClient client = n.serviceClient<hector_uav_msgs::EnableMotors>("enable_motors");

  hector_uav_msgs::EnableMotors srv;
  srv.request.enable = true;

  if(client.call(srv))
  {
    ROS_INFO("Successfully enabled motors?: %s", srv.response.success ? "true" : "false");
  }
  else
  {
    ROS_INFO("Failed to call service enable_motors");
    return 1;
  }

  actionlib::SimpleActionClient<hector_uav_msgs::PoseAction> ac("action/pose", true);

  ROS_INFO("Waiting for action server to start.");
  ac.waitForServer();

  ROS_INFO("Action server started, sending goal.");

  
  hector_uav_msgs::PoseGoal goal;
  goal.target_pose.header.frame_id = "world";
  goal.target_pose.pose.position.x = 0;
  goal.target_pose.pose.position.y = 0;
  goal.target_pose.pose.position.z = 1;

  goal.target_pose.pose.orientation.x = 0;
  goal.target_pose.pose.orientation.y = 0;
  goal.target_pose.pose.orientation.z = 0;
  goal.target_pose.pose.orientation.w = 1;

  ros::Time stamp = ros::Time::now();
  goal.target_pose.header.stamp.sec = stamp.sec;
  goal.target_pose.header.stamp.nsec = stamp.nsec;


  ROS_INFO("Send takeoff goal");
  ac.sendGoal(goal);

  bool finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

  if(finished_before_timeout)
  {
  	actionlib::SimpleClientGoalState state = ac.getState();
  	ROS_INFO("Takeoff finished: %s", state.toString().c_str());
  }
  else
  	ROS_INFO("Takeoff did not finish before the time out.");

  goal.target_pose.pose.position.x = 2;
  goal.target_pose.pose.position.y = 2;
  goal.target_pose.pose.position.z = 2;
  stamp = ros::Time::now();
  goal.target_pose.header.stamp.sec = stamp.sec;
  goal.target_pose.header.stamp.nsec = stamp.nsec;

  ac.sendGoal(goal);

  finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

  if(finished_before_timeout)
  {
  	actionlib::SimpleClientGoalState state = ac.getState();
  	ROS_INFO("Action finished: %s", state.toString().c_str());
  }
  else
  	ROS_INFO("Action did not finish before the time out.");

	
  return 0;
}

File: hector_quadrotor/hector_quadrotor_navigation/src/quadrotor_navigation.py
#!/usr/bin/env python
import rospy
import time
import random
from hector_uav_msgs.srv import EnableMotors
from geometry_msgs.msg import PoseStamped
import actionlib
import hector_uav_msgs.msg

def hector_pose_client():
     # First enable motor service
     rospy.wait_for_service("/enable_motors")
     enabler1 = rospy.ServiceProxy("/enable_motors", EnableMotors)
     resp1 = enabler1(True)

     rospy.loginfo("Creating Action Client.")
     client = actionlib.SimpleActionClient('/action/pose', hector_uav_msgs.msg.PoseAction)
     rospy.loginfo("Client created.")

     # This is where the program seems to hang even though I assumed hector would automatically run the action server
     client.wait_for_server()

     # Create a random goal
     g = hector_uav_msgs.msg.PoseGoal()
     g.target_pose.header.frame_id = 'world'
     g.target_pose.pose.position.x = 0
     g.target_pose.pose.position.y = 1
     g.target_pose.pose.position.z = 3

     rospy.loginfo("Sending goal")
     client.send_goal(g)
     rospy.loginfo("Waiting for result")
     client.wait_for_result()
     return(client.get_result())

if __name__ == "__main__":
     try:
         rospy.init_node('drone_explorer')
         result = hector_pose_client()
         rospy.loginfo("Client navigated")
     except rospy.ROSInterruptException:
         print("program interrupted before completion")